### DISCUSSION TITLE
How to define material properties in 2-phase double-point method?

### DISCUSSION THREAD

**Discussion by qiuqiubing** [Anura3D/Anura3D_OpenSource#66] [Q&A] [answered]
Hi, dear all,
I'm using the 2-phase double-point formulation to simulate the column collapse problem (as shown in the picture below) described in the article "Modelling soil-water interaction with the MPM. Evaluation of single-point and double-point formulations". There's a question confusing me for some time. How to define the material properties of both solid and liquid for the same space area? Should I choose "saturated material-fully coupled" or define material properties for solids and liquids separately? I didn't find any instructions for the 2-phase double-point method in the tutorial. Hope for your answers.
![1676857820948](https://user-images.githubusercontent.com/48994907/219991087-d4333200-31a4-4f49-8d50-2cc7c6044522.png)

**Comment by FrancescaCeccato** [comment]
Dear qiuqiubing,
for the materials:
1. if the material is saturated soil or dry soil, then assign _saturated material - fully coupled_ (this is the option you have to use for your example)
2.  if it is pure water, then use _liquid_
When you assign the number of material points (MPs) per element you have to assign the number of solid MPs and liquid MPs. The code automatically recognize that you are using double point formulation if you use both solid and liquid MPs.
For the boundary conditions you have to prescribe fixities and loads to liquid and solid separately.

**Reply by qiuqiubing** [reply]
Thank you very much for your reply. I understand how to define material properties in a double-point method.

My model still has some problems. The material is saturated soil. The number of material points and boundary fixities are assigned for solid and liquid respectively. The parameters required for the double-point method are specified in the calculated data. 
![solid_fixities](https://user-images.githubusercontent.com/48994907/221333459-92ffa63d-cd58-4109-b083-c71186714865.png)
![liquid_fixities](https://user-images.githubusercontent.com/48994907/221333460-1c204446-fe79-40f9-b723-4605a3419b07.png)
![material_point](https://user-images.githubusercontent.com/48994907/221333461-da788196-8ef6-4a21-bdaa-45ec71db88b5.png)
![calculation_data](https://user-images.githubusercontent.com/48994907/221333476-b909bf99-1a72-4142-945c-46541a9d59de.png)

The error information of the quasi-static analysis step is as follows:
![error](https://user-images.githubusercontent.com/48994907/221333481-ba742054-fc13-495a-a3d5-0bccb91e62c2.png)

What might be wrong with the model?

**Comment by murphy-john** [comment, marked as answer]
You define the material the same way you do in the single-point method in the "Materials" screen. I've only used the "Saturated Material - Fully Coupled" material type with the double-point method. I don't know how other material models would work.

What's different in the model creation phase is you'll need to set the numbers of liquid and solid material points in the "Material Point Specification" screen. Change from the single point to the double point in the drop down menu and then input the number of each material point per element you want in the text entry boxes. The code will detect you've selected more than one type of material and assign the material properties to the appropriate material point type.

**Reply by qiuqiubing** [reply]
Thank you, Murphy! Now I understand how to define material properties in a double-point method.

**Comment by PetitPipi** [comment]
Hi qiuqiubing and Team,

Can you guys show the material properties you defined?
I'm running the same example, following the article that qiuqiubing mentioned above, but I keep getting this error: 'WaveSpeed_c1 is not a number or infinity'.
Another discussion indicated that the issue is related to material properties defined in the GOM file, so I'd love to see the material properties you defined, therefore I can find out where I did wrong.  Currently, I'm using the parameters given in the article. 

Best,
P

**Reply by GBarbosaS** [reply]
Hi, i'm having the same problem, Did you figure out how to fix this?

**Reply by PetitPipi** [reply]
Hi, It's been so long I forgot where the issue was, but I highly recommend you use the debug function in Visual Studio, which will help you to locate the problematic lines.

**Reply by jiawei6688** [reply]
> Hi, i'm having the same problem, Did you figure out how to fix this?

**Comment by mikael-onsilva** [comment]
@qiuqiubing, were you able to re-create that simulation? Can you explain wich materials, boundary conditions, etc, did you use? And/or share with us the model itself?

Thanks in advance.
